clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -577,13 +577,17 @@ public void logUnused() {
                 configProviderInstances.put(entry.getKey(), provider);
             } catch (ClassNotFoundException e) {
                 log.error("ClassNotFoundException exception occurred: " + entry.getValue());
-                throw new ConfigException("Invalid config:" + entry.getValue() + " ClassNotFoundException exception occurred", e);
+                throw new ConfigException(providerClassProperty(entry.getKey()), entry.getValue(), "ClassNotFoundException exception occurred");
```

Comment:
What if the class that's not found isn't the config provider but some dependency of it? Because the CNFE message isn't logged nor passed to the `ConfigException` this info would be lost.

Reply:
Good point; I've updated the log line above to include the entire stack trace. It'd be nice if there were a variant of `ConfigException` that took in a cause argument but there is none at the moment and it's probably not worth it to add one just for this use case.